src/realm/sync/client.cpp

```diff
@@ -1967,7 +1969,7 @@ void ClientImpl::Connection::resume_active_sessions()
 
 void ClientImpl::Connection::on_idle()
 {
-    logger.debug("Destroying connection object");
+    logger.debug(util::LogCategory::session, "Destroying connection object");
```

Comment:
connection can serve multiple session. Maybe introduce one more category under client for Connection? Otherwise just 'client' would be appropriate i think. @michael-wb what are you thought?